tests/integration/Core/Checkout/Cart/Order/OrderRecalculationControllerTest.php

```diff
@@ -0,0 +1,160 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Integration\Core\Checkout\Cart\Order;
+
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Checkout\Cart\Price\Struct\CalculatedPrice;
+use Shopware\Core\Checkout\Cart\Tax\Struct\CalculatedTaxCollection;
+use Shopware\Core\Checkout\Cart\Tax\Struct\TaxRuleCollection;
+use Shopware\Core\Checkout\Order\OrderCollection;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Test\TestCaseBase\AdminApiTestBehaviour;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\Test\Integration\Traits\OrderFixture;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class OrderRecalculationControllerTest extends TestCase
+{
+    use AdminApiTestBehaviour;
+    use IntegrationTestBehaviour;
+    use OrderFixture;
+
+    /**
+     * @var EntityRepository<OrderCollection>
+     */
+    private EntityRepository $orderRepository;
+
+    private Context $context;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->orderRepository = static::getContainer()->get('order.repository');
+
+        $this->context = Context::createDefaultContext();
+    }
+
+    public function testPrimaryDeliveryAndTransactionIdStayTheSame(): void
+    {
+        $orderId = Uuid::randomHex();
+        $primaryOrderDeliveryId = Uuid::randomHex();
+        $primaryTransactionId = Uuid::randomHex();
+        $versionId = Uuid::randomHex();
+
+        $this->createOrder($orderId, $primaryOrderDeliveryId, $primaryTransactionId, $versionId);
+
+        $browser = $this->getBrowser();
+        $browser->request('POST', \sprintf('/api/_action/order/%s/recalculate', $orderId));
+
+        $this->context->assign(['versionId' => $versionId]);
+
+        /**
+         * @var OrderEntity $order
```

Comment:
please do not use var annotations. Especially in tests you can simply assert the type. also see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md